### PR TITLE
Add performance timing logging to Hosted js modules

### PR DESCRIPTION
**Pull request by rich-nguyen**
This alters the commercial bootstrap behaviour for all commercial modules which use custom timing. These modules all share the same pattern; they use custom timing to measure their execution time because the work they do is done asynchronously (usually through the gpt command buffer), and, crucially, this asynchronous work does _not_ need to be waited for in order for execution of dependent modules to begin.

Another way to think of this is that these modules add work to the queue, and the act of adding it to the queue is the only part of the module that must be completed by the bootstrap. The queued work itself is ordered, and safe to execute outside of the bootstrap's promise handling.

So, the promise that encloses this kind of async work only serves to tell the bootstrap that the module has fully completed, and this information is *only useful in the context of hosted modules, whose pages have no MPUs, which causes the performance logging POST to be skipped.

(*only - there are some other situations we could consider in the future. Any page which has no MPUs served through DFP will not be performance logged).

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/hosted/video.js [hunk 1]
```diff
@@ -60,84 +60,90 @@ define([
...
+            require(['bootstraps/enhanced/media/main'], function() {
+                require(['bootstraps/enhanced/media/video-player'], function (videojs) {
+                    var $videoEl = $('.vjs-hosted__video');
+                    var $inlineVideoEl = $('video');
+                    var $youtubeIframe = $('.js-hosted-youtube-video');
+
+                    if ($youtubeIframe.length === 0 && $videoEl.length === 0) {
+                        if ($inlineVideoEl.length === 0) {
+                            // halt execution
+                            performanceLogging.moduleEnd(moduleName);
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/commercial/modules/hosted/video.js]: view this change with `?w=1` to see what i've done here.

File: static/src/javascripts/projects/commercial/modules/hosted/video.js [hunk 2]
```diff
@@ -60,84 +60,90 @@ define([
...
+                            resolve();
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/hosted/video.js]: `resolve` does not stop execution, that is why there needs to be a `return` statement.

File: static/src/javascripts/projects/commercial/modules/hosted/onward.js [hunk 3]
```diff
@@ -31,10 +33,12 @@ define([
...
                 })
-                .then(function () {
-                    performanceLogging.moduleEnd(moduleName);
-                });
+        } else {
+            fetchPromise = Promise.resolve();
         }
-        return Promise.resolve();
+
+        return fetchPromise.then(function () {
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/commercial/modules/hosted/onward.js]: yes, hosted gallery and hosted onward both need to return a promise to bootstrap. The promise will only be used to wait until all performance `moduleEnd` calls have been made. At that point, I can report the perf data.

File: static/src/javascripts/projects/commercial/modules/hosted/video.js (hunk 2 above)
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/hosted/video.js]: (although I would remove that whole test, `$videoEl.each` and `youtubeIframe.each` will handle the `length === 0` naturally)

File: static/src/javascripts/projects/commercial/modules/hosted/onward.js (hunk 3 above)
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/hosted/onward.js]: (in fact, the contrary, why _not_ return the `loadCssPromise` in gallery?

Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/hosted/onward.js]: oops, just saw your comment :zipper_mouth_face:

**Review by regiskuckaertz — APPROVED** [review]
Great! 👍

File: static/src/javascripts/projects/commercial/modules/hosted/onward.js (hunk 3 above)
Comment (rich-nguyen) [inline static/src/javascripts/projects/commercial/modules/hosted/onward.js]: I do!

File: static/src/javascripts/projects/commercial/modules/hosted/video.js (hunk 2 above)
Comment (rich-nguyen) [inline static/src/javascripts/projects/commercial/modules/hosted/video.js]: 👍

File: static/src/javascripts/projects/commercial/modules/hosted/video.js [hunk 4]
```diff
@@ -60,84 +60,90 @@ define([
...
-                            }
-                        }
+                    $youtubeIframe.each(function(el){
+                        hostedYoutube.init(el);
                     });
-                });
 
-                $youtubeIframe.each(function(el){
-                    hostedYoutube.init(el);
+                    performanceLogging.moduleEnd(moduleName);
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/commercial/modules/hosted/video.js]: i think that would be tidier, yes

**Comment by rich-nguyen** [comment]
[Improve commercial loading order](https://trello.com/c/E8ojiTwb/260-improve-commercial-loading-order)

**Pull request merged**